src/path-planner.cc

```diff
@@ -149,9 +163,13 @@ namespace hpp {
 	  PathValidationReportPtr_t report;
           bool pathValid = pathValidation->validate (projPath, false, validPath,
 						     report);
-          if (pathValid && validPath->length() > 0) {
+          if (pathValid && validPath->timeRange ().second !=
+              path->timeRange ().first) {
             roadmap ()->addEdge (initNode, *itn, projPath);
-            roadmap ()->addEdge (*itn, initNode, projPath->reverse());
+            interval_t timeRange = projPath->timeRange ();
+            roadmap ()->addEdge (*itn, initNode, projPath->extract
+                (interval_t (timeRange.second,
+                             timeRange.first)));
```

Comment:
`projPath->reverse()` would be more readable.

Reply:
Done in https://github.com/humanoid-path-planner/hpp-core/pull/117/commits/d93434ef4f296275d953cc820701eeda9b85c224